src/sv_port.rs

```diff
@@ -220,6 +224,68 @@ fn port_signedness_ansi(
     }
 }
 
+fn port_packeddim_ansi(
+    m: &sv_parser::AnsiPortDeclaration,
+    syntax_tree: &SyntaxTree,
+) -> Vec<SvPackedDimension> {
+    let mut ret: Vec<SvPackedDimension> = Vec::new();
+
+    for node in m {
+        match node {
+            RefNode::PackedDimensionRange(x) => {
+                let mut _div_found: bool = false;
+                let mut upper = String::new();
+                let mut lower = String::new();
+
+                let range = unwrap_node!(x, ConstantRange);
+                match range {
+                    Some(RefNode::ConstantRange(sv_parser::ConstantRange { nodes })) => {
+                        let (u, _, l) = nodes;
+                        for sub_node in u {
+                            match sub_node {
+                                RefNode::BinaryOperator(_) => {
+                                    upper.push_str(&symbol(sub_node, syntax_tree).unwrap())
+                                }
+                                RefNode::Identifier(_) => {
+                                    upper.push_str(&identifier(sub_node, syntax_tree).unwrap())
+                                }
+                                RefNode::Number(_) => {
+                                    upper.push_str(&number(sub_node, syntax_tree).unwrap())
+                                }
+                                _ => (),
+                            }
+                        }
```

Comment:
I think this could be done more safely with something like `format!("{}", u)` instead of the for/match.
Iterating over sub-nodes and extracting only some of their strings feels like it might miss edge cases.

Reply:
Changed it to a new implementation that relies on "all_tokens" function.